### DISCUSSION TITLE
Relative permeability error

### DISCUSSION THREAD

**Discussion by ruchajk2597** [PMEAL/OpenPNM#2580] [Q&A]
Hello Community, 

I am running the relative permeability algorithm and getting the following error. When I remove the snwp the algorithm runs but it means I do not have any pots because there are no points. It says that by default it takes Snwp=100 but tha is not the case. If I manually give some number it throws an error. 
I am writing my thesis and this is a very important result for that. It would be great if someone can help me debug the code. 

Below is my code snippet and error. 

CODE
rp = op.algorithms.metrics.RelativePermeability(network=pn)
rp.setup(invading_phase='n2',invasion_sequence='pore.invasion_sequence')
rp.run(Snwp_num=100)

ERROR

AttributeError                            Traceback (most recent call last)
Input In [124], in <cell line: 4>()
      2 rp.setup(invading_phase='n2',invasion_sequence='pore.invasion_sequence')
      3 rp.setup(Snwp_num=100)
----> 4 rp.run(Snwp_num=100)

File C:\ProgramData\Anaconda3\lib\site-packages\openpnm\algorithms\metrics\RelativePermeability.py:324, in RelativePermeability.run(self, Snwp_num)
    322     Snwp_num = self.settings['Snwp_num']
    323 net = self.project.network
--> 324 K_dir = set(self.settings['flow_inlets'].keys())
    325 for dim in K_dir:
    326     flow_pores = [net.pores(self.settings['flow_inlets'][dim]),
    327                   net.pores(self.settings['flow_outlets'][dim])]

AttributeError: 'NoneType' object has no attribute 'keys'

Thank you in advance. 

Best Regards,
Rucha

**Comment by jgostick** [comment]
Hi @ruchajk2597 
In V3, which is nearly ready, we have removed the 'metrics' folder, in favor of directing people to detailed examples on how the compute the specific metrics of interest.  [This example](https://openpnm.org/examples/applications/relative_permeability.html) is a bit of a mess at the moment, but gives you the idea.  The problem with the metrics functions is that they have to do too much, make too many assumptions, etc.

**Comment by ruchajk2597** [comment]
Hi,

Thank you for the response.
I was able to solve the error and run the relative permeability code.

Best Regards,
Rucha

Von: jgostick ***@***.***>
Gesendet: Dienstag, 27. September 2022 15:33
An: PMEAL/OpenPNM ***@***.***>
Cc: Kunte, Rucha ***@***.***>; Mention ***@***.***>
Betreff: Re: [PMEAL/OpenPNM] Relative permeability error (Discussion #2580)


Hi @ruchajk2597<https://github.com/ruchajk2597>
In V3, which is nearly ready, we have removed the 'metrics' folder, in favor of directing people to detailed examples on how the compute the specific metrics of interest. This example<https://openpnm.org/examples/applications/relative_permeability.html> is a bit of a mess at the moment, but gives you the idea. The problem with the metrics functions is that they have to do too much, make too many assumptions, etc.
